web/src/composebox_typeahead.ts

```diff
@@ -1180,6 +1183,32 @@ export function content_typeahead_selected(
     return beginning + rest;
 }
 
+function will_produce_broken_stream_topic_link(topic_name: string): boolean {
+    return /(\*+)|(.*`.*`.*)/.test(topic_name);
+}
+
+function get_stream_name(syntax: string): string {
+    const start = syntax.lastIndexOf("#**");
+    const end = syntax.lastIndexOf(">");
+    return syntax.slice(start + 3, end);
+}
+
+function url_syntax(stream_name: string, topic_name: string): string {
+    const stream = stream_data.get_sub(stream_name);
+    const stream_id = stream?.stream_id ?? -1;
+    return `[#${stream_name}>${topic_name}](${hash_util.by_stream_topic_url(stream_id, topic_name)}) `;
```

Comment:
Where does -1 come from here?

I believe that will end up producing a link that's broken, and doesn't even have the stream name the user typed. In general, when we don't have a meaningful value to supply somewhere, it's best not to make up a bogus value like -1 — instead, either do something that explicitly means "there's no known value", or if that's not possible then throw an exception.

(And if throwing an exception feels like it'd be a real problem for users, then that's a sign we need to do some more work so as to actually handle the case where we don't have a real value: either get the value after all, or gracefully handle saying "there's no known value", or something else.)

Reply:
Since this code path is only accessed through typeaheads, it is improbable that `stream_id` would ever be `-1`. I think we can throw an exception.

Reply:
Yes, an exception sounds correct to me.